packages/ckeditor5-engine/src/view/observer/selectionobserver.ts

```diff
@@ -232,6 +232,10 @@ export default class SelectionObserver extends Observer {
 
 		const domSelection = domDocument.defaultView!.getSelection()!;
 
+		if ( env.isGecko && this._isObjectRestricted( domSelection.getRangeAt( 0 ).startContainer ) ) {
```

Comment:
How about we `s/_isObjectRestricted/isGeckoRestrictedObject` and we move the `env.isGecko` check as the first line of the helper? IMO this would improve the readability of `_handleSelectionChange()` because the code would be as much natural language as possible.